src/main/kotlin/org/rust/cargo/project/model/impl/CargoSyncTask.kt

```diff
@@ -421,4 +427,25 @@ private fun <T, R> BuildProgress<BuildProgressDescriptor>.runWithChildProgress(
     }
 }
 
+/**
+ * This cache is really used *only* in unit tests.
+ *
+ * We assume that `sysroot`, a `list of targets`, `--print cfg` and `stdlib`
+ * are not changed until the version of rustc is changed, so we can cache these values for all tests.
+ * The cache significantly speeds up heavy tests with a full toolchain (`RsWithToolchainTestBase`)
+ */
+private object UnitTestRustcCache {
```

Comment:
Probably, it may be useful in other plugin parts so I'd extract it into a separate file as public class. But see general suggestions for this PR first